aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,94 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void axis_check(IntArrayRef& axis, int64_t bound) {
+  std::unordered_set<int64_t> dimensions;
+  for (const auto i : c10::irange(axis.size())) {
+    int64_t pos_axis  = axis[i] >= 0 ? axis[i] : axis.size()+axis[i];
+    TORCH_CHECK( dimensions.find(pos_axis) == dimensions.end(), "repeated axis");
+    TORCH_CHECK( (axis[i] < bound) && (axis[i] >= (-1*bound)), "axis " + std::to_string(axis[i]) +" is out of bounds for array of dimension " + std::to_string(bound)  );
+    dimensions.insert( pos_axis);
+  }
+}
+
+std::vector<Tensor> gradient(const Tensor& self,
+                             const Tensor& spacing,
+                             IntArrayRef axis,
+                             int64_t edge_order) {
+  axis_check(axis, self.dim());
+  TORCH_CHECK( spacing.size(0) == axis.size(), "invalid number of argumentsxsx");
+  TORCH_CHECK( edge_order == 1, "Values of edge_order larger than 1 does not get supported supported.");
+
```

Comment:
change spacing to tensorlist instead of tensor.. 
check akk tensors are 1-dimension and also all devices are same. aslo sizes of spacing match dimension size.